mantidproject/mantid
# Fix About not Launching on New Version if Nightly Installed and Refactor Version Information

**Description of work.**
Made changes to the version information files and python interface to allow a python `sys` style `version_info` struct. 
The major, minor, and patch numbers can now be extracted from this struct as and when they are useful, rather than having to deconstruct a string. 

**To test:**
**To check the new struct is working correctly:**
1. Import the new function (`from mantid.kernel import version_info`)
2. Check the major, minor, and patch values (`version_info().major` etc.)

**To test the new about menu changes:**
1. Obseve the original issue:
    a. Launch the Mantid 6.2 Release of workbench
    b. Check the "Do Not Show Until Next Release" checkbox
    c. Close Workbench
    d. Lauch a 6.2 nightly or build (NOT this PR)
    e. Launch this workbench
    f. The about page appears when it should not.
2. Observe the fix:
    a. Launch the new build of this PR to set the QSetting and check "Do Not Show Until Next Release" checkbox
    b. Close workbench
    c. Launch the Mantid 6.2 Release of workbench and check "Do Not Show Until Next Release" checkbox
    d. Close workbench
    e. Launch the workbench for this PR
    f. The about page should no longer appear as the major and minor version is the same


Fixes #32653 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
